File: projects/gmail-outreach/cron-enrich-536pm.js
const fs = require('fs');

// Read the current PE data
const data = JSON.parse(fs.readFileSync('current-pe-data.json', 'utf8'));

// Find firms needing enrichment
const needsEnrichment = data.filter(row => {
  const name = (row['Contact Name'] || '').trim();
  const email = (row.Email || '').trim();
  
  // Empty contact name OR generic/empty email
  const emptyName = !name || name === '';
  const genericEmail = email && (
    email.startsWith('info@') || 
    email.startsWith('sales@') || 
    email.startsWith('ir@') ||
    email.startsWith('contact@') ||
    email.startsWith('hello@') ||
    email.startsWith('admin@') ||
    email.startsWith('support@')
  );
  const emptyEmail = !email || email === '';
  
  return emptyName || genericEmail || emptyEmail;
});

console.log(`Found ${needsEnrichment.length} firms needing enrichment`);
console.log('\nFirst 15 firms to enrich:\n');

needsEnrichment.slice(0, 15).forEach((firm, idx) => {
  console.log(`${idx + 1}. ${firm['Company Name']}`);
  console.log(`   Website: ${firm.Website || 'N/A'}`);
  console.log(`   Current Contact: ${firm['Contact Name'] || '[EMPTY]'}`);
  console.log(`   Current Email: ${firm.Email || '[EMPTY]'}`);
  console.log(`   Status: ${firm.Status || 'N/A'}`);
  console.log('');
});

// Save the enrichment targets
fs.writeFileSync('enrich-targets-536pm.json', JSON.stringify(needsEnrichment.slice(0, 15), null, 2));
console.log('\nSaved enrichment targets to enrich-targets-536pm.json');

File: projects/gmail-outreach/cron-enrich-536pm.py
import json

# Read the current PE data
with open('current-pe-data.json', 'r', encoding='utf-8') as f:
    data = json.load(f)

# Find firms needing enrichment
needs_enrichment = []
for row in data:
    name = (row.get('Contact Name') or '').strip()
    email = (row.get('Email') or '').strip()
    
    # Empty contact name OR generic/empty email
    empty_name = not name
    generic_email = email and any(email.startswith(prefix + '@') for prefix in 
        ['info', 'sales', 'ir', 'contact', 'hello', 'admin', 'support'])
    empty_email = not email
    
    if empty_name or generic_email or empty_email:
        needs_enrichment.append(row)

print(f"Found {len(needs_enrichment)} firms needing enrichment")
print("\nFirst 15 firms to enrich:\n")

for idx, firm in enumerate(needs_enrichment[:15]):
    print(f"{idx + 1}. {firm.get('Company Name', 'N/A')}")
    print(f"   Website: {firm.get('Website', 'N/A')}")
    print(f"   Current Contact: {firm.get('Contact Name') or '[EMPTY]'}")
    print(f"   Current Email: {firm.get('Email') or '[EMPTY]'}")
    print(f"   Status: {firm.get('Status', 'N/A')}")
    print()

# Save the enrichment targets
with open('enrich-targets-536pm.json', 'w', encoding='utf-8') as f:
    json.dump(needs_enrichment[:15], f, indent=2, ensure_ascii=False)

print("\nSaved enrichment targets to enrich-targets-536pm.json")
